clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -2743,9 +2746,21 @@ int main(int Argc, char **Argv) {
       if (Error Err = sycl::mergeSYCLBIN(*FilesOrErr, Args))
         reportError(std::move(Err));
     } else {
-      // Run the host linking job with the rendered arguments.
-      if (Error Err = runLinker(*FilesOrErr, Args))
-        reportError(std::move(Err));
+      if (Args.hasArg(OPT_sycl_device_link)) {
+        // Skip host linker if -sycl-device-link option is set.
+        // Just copy the output of device linking and wrapping action.
+        if (FilesOrErr->size() > 1) {
+          reportError(createStringError(
+              "Expect single output from the device linker."));
+        } else {
+          if (Error Err = sycl::copyFileToFinalExecutable((*FilesOrErr)[0]))
+            reportError(std::move(Err));
+        }
```

Comment:
Don't we need to handle the case when `FilesOrErr->size() == 0`?
```suggestion
        if (FilesOrErr->size() == 1) {
          if (Error Err = sycl::copyFileToFinalExecutable((*FilesOrErr)[0]))
            reportError(std::move(Err));
        } else {
          reportError(createStringError(
              "Expect single output from the device linker."));
        }
```

Reply:
I suggested the version that handles all sizes.